From da7bbea751fde7063a492fb832a659f6abb0ce93 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Sat, 14 Sep 2024 08:41:49 +0300
Subject: [PATCH] Revert "[FSW-2704] fs1: temp spi dma fix for auto mode"

This change made when we dont have provision for
new SPI DMA for copying data more than 64kb.
After psoc tag 1.15, we have support for new SPI
DMA. Hence reverting this change.

This reverts commit bcc75b259a53e2f438e294162abb05553dc8b42d.

Change-Id: Ie3cc9ab01d2425df89e7e87053b1595bd18a9fd6
---

diff --git a/arch/fs1/arc/dev_dma.c b/arch/fs1/arc/dev_dma.c
index 830e5cc..48d3f1f 100644
--- a/arch/fs1/arc/dev_dma.c
+++ b/arch/fs1/arc/dev_dma.c
@@ -29,8 +29,8 @@
 }
 
 void dma_copy_image_internal(uint16_t image_size, uint32_t flash_offset,
-		uint64_t dest_offset, uint8_t fw_sram_dst, __unused uint8_t restart_crc,
-		__unused uint8_t mask_crc)
+		uint64_t dest_offset, uint8_t fw_sram_dst, uint8_t restart_crc,
+		uint8_t mask_crc)
 {
 	uint32_t reg_spi_dma_ctrl_s =  0;
 	uint32_t spi_dma_dst_addr_l = DMA_ADDRESS_LOW(dest_offset);
@@ -50,10 +50,10 @@
 	reg_spi_dma_ctrl_s |= fw_sram_dst << PARC_GLOBAL_CONF_SPI_DMA_CTRL_DST_SRAM_S;
 
 	/* 0: continue crc calc 1: restart crc*/
-	// reg_spi_dma_ctrl_s |= restart_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_CRC_RESTART_S;
+	reg_spi_dma_ctrl_s |= restart_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_CRC_RESTART_S;
 
 	/* 0: pass crc error to status 1: mask crc error */
-	// reg_spi_dma_ctrl_s |= mask_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_MASK_CRC_ERR_S;
+	reg_spi_dma_ctrl_s |= mask_crc << PARC_GLOBAL_CONF_SPI_DMA_CTRL_MASK_CRC_ERR_S;
 
 	/* # of 4bytes to DMA copy */
 	reg_spi_dma_ctrl_s |= image_size_4b << PARC_GLOBAL_CONF_SPI_DMA_CTRL_MEM_SIZE_S;
